Tools/upload_wiki_icons.py

```
@@ -0,0 +1,131 @@
+def list_existing_filenames(base_url: str, token: str, folder_id: int) -> set[str]:
+    data = graphql(
+        base_url,
+        token,
+        f'{{ assets {{ list(folderId: {folder_id}, kind: IMAGE) {{ filename }} }} }}',
+    )
+    return {item["filename"] for item in data["assets"]["list"]}
+
+
+def upload_icon(base_url: str, token: str, folder_id: int, icon_path: Path) -> tuple[str, str]:
+    wiki_name = icon_path.name.lower()
+    result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)